lib/pbench/test/unit/server/query_apis/test_datasets_update.py

```diff
@@ -240,14 +240,17 @@ def test_query_owner_publish(
         is_admin = build_auth_header["header_param"] == HeaderTypes.VALID_ADMIN
         if not HeaderTypes.is_valid(build_auth_header["header_param"]):
             expected_status = HTTPStatus.UNAUTHORIZED
-        elif not bool(query_json):
+        elif not query_json:
             expected_status = HTTPStatus.BAD_REQUEST
-        elif not is_admin and owner:
+            ds_name = "drb"
+        elif (not is_admin and owner) or (
+            build_auth_header["header_param"] == HeaderTypes.VALID and owner == None
+        ):
```

Comment:
If the token is valid, then the requestor is `"drb"` (who is not an Admin), so the two times when we should get `FORBIDDEN` are when `owner` is specified and when `ds_name` is `"test"`.  So, perhaps something more like this:
```
        elif not is_admin and (owner or ds_name == "test"):
```

Reply:
That is, `build_auth_header["header_param"] == HeaderTypes.VALID and owner == None` should not necessarily be forbidden, depending on what `ds_name` and `access` are...so my proposed test should hold water better.